File: memory.py
import psutil
import os
import time

maxCounter = 1000000
maxRounds = 100
counters = []

min = 8 * 1024 * 1024 * 1024 # 8G
max = 0
memory = []

p = psutil.Process( os.getpid() )

start = time.time()

for round in range(maxRounds):
    for counter in range(maxCounter):
        if round == 0:
            counters[counter] = counter
        else:
            counters.append(counter)

    m = p.memory_info()[1]

    if m < min: min = m
    if m > max: max = m

    memory.append(m)

    # print(str(round) + ': ' + str(m))

print(time.time() - start)
print(min)
print(sum(memory)/len(memory))
print(max)

File: memory.go
package main

import "fmt"
import "time"
import "runtime"

type IntWrapper struct {
	v int
}

func main() {
	
	const size = 1000000
	const rounds = 100
	
	// var numbers [size]int
	var numbers [size]IntWrapper
	
	var min uint64 = 8 * 1024 * 1024 * 1024
	var max uint64 = 0

	memory := make([]uint64, rounds)
	
	var mem runtime.MemStats
	
	start := time.Now()
	
	for round := 0; round < rounds; round++ {
	
		for counter := 0; counter < size; counter++ {
			numbers[counter] = IntWrapper { counter }
		}
		
		runtime.ReadMemStats(&mem)
		
		if mem.Alloc < min {
			min = mem.Alloc
		}
		
		if mem.Alloc > max {
			max = mem.Alloc
		}
		
		memory[round] = mem.Alloc
		
		// fmt.Printf("round: %d\n", mem.Alloc)
		
	}
	
	fmt.Printf("time: %s\n", time.Since(start))
	fmt.Printf("min: %d\n", min)
	var tmp float64 = 0
	for i := 0; i < rounds; i++ {
		tmp += float64(memory[i])
	}
	tmp = tmp / rounds
	fmt.Printf("avg: %f\n", tmp)
	fmt.Printf("max: %d\n", max)
	
}